### PR TITLE
Bay/-tg- Meteor port

### PR DESCRIPTION
**Pull request by tigercat2000**
This ports Bay's implementation (Baystation12/Baystation12#9694) of -tg-'s meteors.

Things changed:
 - Meteor events merged into one, difference between moderate and major event is types of meteor.
 - New meteor types, and proper integration of meteor types.
  - Radioactive meteors, cause toxin damage to mobs that it lands near.
  - Tunguska, only seen in major event, causes a massive amount of damage to the station.
  - Proper flaming meteors, which don't cause fires.. yet.

Quote from the baystation PR:
> Meteors also appear to be more accurate, missing the station much less seldom (and usually at least hit the solar wings instead).

### CODE REVIEW

**Comment by TheDZD** [comment]
Might want to merge #1353 before this, so explosions with a flame range work properly.

**Comment by Allfd** [comment]
Yes please.

It looks exciting.

File: code/game/gamemodes/events/clang.dm [hunk 1]
```diff
@@ -34,7 +34,7 @@ In my current plan for it, 'solid' will be defined as anything with density == 1
 
 		else if (istype(clong, /mob))
 			if(clong.density || prob(10))
-				clong.meteorhit(src)
+				clong.ex_act(3)
```
Comment (Fox-McCloud) [inline code/game/gamemodes/events/clang.dm]: ex_act(2)

File: code/game/gamemodes/events/dust.dm [hunk 2]
```diff
@@ -99,7 +99,7 @@ The "dust" will damage the hull of the station causin minor hull breaches.
 				playsound(src.loc, 'sound/effects/meteorimpact.ogg', 40, 1)
 
 				if(ismob(A))
-					A.meteorhit(src)//This should work for now I guess
+					A.ex_act(3)//This should work for now I guess
```
Comment (Fox-McCloud) [inline code/game/gamemodes/events/dust.dm]: A.ex_act(strength)

File: code/modules/projectiles/projectile/special.dm [hunk 3]
```diff
@@ -75,7 +75,7 @@
 		if(src)//Do not add to this if() statement, otherwise the meteor won't delete them
 			if(A)
 
-				A.meteorhit(src)
+				A.ex_act(3)
```
Comment (Fox-McCloud) [inline code/modules/projectiles/projectile/special.dm]: A.ex_act(2)

**Comment by Fox-McCloud** [comment]
I'll quote @RemieRichards  on this one:

>>not porting my spooky meteors too

>cri

File: code/game/gamemodes/events/clang.dm (hunk 1 above)
Comment (tigercat2000) [inline code/game/gamemodes/events/clang.dm]: that's not consistent with bay

Comment (Fox-McCloud) [inline code/game/gamemodes/events/clang.dm]: But it is consistent with TG, where this *came* from and was designed.

File: code/game/gamemodes/events/dust.dm (hunk 2 above)
Comment (tigercat2000) [inline code/game/gamemodes/events/dust.dm]: doesn't paticularly matter, it's not enabled, but alright

**Comment by Fox-McCloud** [comment]
Putting DNR on this until: https://github.com/ParadiseSS13/Paradise/pull/1353 is merged in, as s "Just in case someone doesn't notice explosions need to come first".

**Comment by tigercat2000** [comment]
@TheDZD Yes, I didn't touch meteor mode. It will always spawn waves of 'normal' meteors.

**Comment by Fox-McCloud** [comment]
May want to make flaming meteors actually uh...flame.

**Comment by tigercat2000** [comment]
@Fox-McCloud Done.

**Pull request merged**